src/app/main.cpp

```diff
@@ -1008,16 +1009,77 @@ int main( int argc, char *argv[] )
     }
   }
 
+  // Create the application. At this point, the profile is not yet selected
+  // But we need the Qt Application to be created to be able to display the
+  // profile selection dialog if needed
+  QgsApplication myApp( argc, argv, myUseGuiFlag, QString(), QStringLiteral( "desktop" ) );
```

Comment:
I believe all the translation handling still needs to occur BEFORE creation of QgsApplication

Reply:
I'm not sure how to do that. User profiles can have different languages, and at this point, the profile is not yet selected, so we cannot fully load the translation. What we could do is install a QTranslator set up with the system language, before displaying the profile dialog. If I'm not mistaken, this would have the added benefit of providing fallback translation in the system language instead of english if the translation in not complete in the profile language (kind of a fringe case, admittedly)

Reply:
All translation must occur before `QgsApplication::init` is called though.
I pushed a commit that install the translators for the system locale (or for the language specified with --lang) BEFORE displaying the profile selection dialog, so that it can be translated. When the profile is selected, the translators for the profile language are installed, and THEN the GUI is created when QgsApplication::init is called. The most recenlty installed translators have priority, so everything seems to work as expected.